frontend/src/libs/packages/store/middlewares/notification-middleware.ts

```diff
@@ -0,0 +1,59 @@
+import { type TypedStartListening } from '@reduxjs/toolkit';
+import {
+  createListenerMiddleware,
+  isFulfilled,
+  isRejected,
+} from '@reduxjs/toolkit';
+
+import { type AsyncThunkConfig } from '~/libs/types/async-thunk-config.type';
+import { NotificationMessage } from '~/packages/notification/libs/enums/notification-message.enum';
+import { NotificationType } from '~/packages/notification/libs/enums/notification-type.enum';
+import { appActions } from '~/slices/app/app.js';
+
+const notificationMiddleware = createListenerMiddleware();
+
+type AppStartListening = TypedStartListening<
+  AsyncThunkConfig['state'],
+  AsyncThunkConfig['dispatch']
+>;
+
+const startAppListening =
+  notificationMiddleware.startListening as AppStartListening;
+
+startAppListening({
+  matcher: isRejected,
+  effect: (action, listenerApi) => {
+    const config = NotificationMessage[action.type];
+    if (config) {
+      void listenerApi.dispatch(
+        appActions.notify({
+          type: config.type,
+          message: config.message,
+        }),
+      );
+    } else {
+      void listenerApi.dispatch(
+        appActions.notify({
+          type: NotificationType.ERROR,
+        }),
+      );
+    }
```

Comment:
You may just use optional chaining with nullish coalescing, instead of if..else block - as you're just passing payload to `notify` method